qubes/api/internal.py

```diff
@@ -361,3 +361,65 @@ async def suspend_post(self):
                         qubes.config.suspend_timeout,
                         "qubes.SuspendPostAll",
                     )
+
+    @qubes.api.method(
+        "internal.dispvm.preload.Create", scope="local", write=True
+    )
+    async def dispvm_preload_create(self):
+        """
+        Method called to create a preloaded DispVM.
+
+        :return:
+        """
+        dispvm_template = self.arg
+        preload_dispvm = dispvm_template.features.get("preload-dispvm", None)
+        ## TODO: should this function receive disposable name as argument
+        ##       or call admin.vm.CreateDisposable?
+        dispvm = TODO
+        if preload_dispvm:
+            preload_dispvm += " " + dispvm
+        else:
+            preload_dispvm = dispvm
+        dispvm_template.features["preload-dispvm"] = preload_dispvm
+        dispvm.features["internal"] = True
+
+    @qubes.api.method(
+        "internal.dispvm.preload.Use", scope="local", write=True
+    )
+    async def dispvm_preload_use(self):
+        """
+        Method called when usage of preloaded DispVM is requested.
+
+        To use the first preloaded disposable, provide as argument the
+        disposable template, else, provide as argument the disposable name.
+
+        :return:
+        """
+        used_dispvm = self.arg
+        if used_dispvm.klass == "DispVM":
+            dispvm_template = getattr(used_dispvm, "template")
+            use_first = False
+        elif (
+            used_dispvm.klass == "AppVM"
+            and getattr(used_dispvm, "template_for_dispvms", False)
+        ):
+            dispvm_template = used_dispvm
+            use_first = True
+        else:
+            raise qubes.exc.QubesException("Invalid argument provided")
+        preload_dispvm = dispvm_template.features.get("preload-dispvm", None)
+        if not preload_dispvm:
+            return
+
+        preload_dispvm = preload_dispvm.split(" ")
+        if use_first:
+            used_dispvm_name = preload_dispvm[1:]
+        else:
+            used_dispvm_name = used_dispvm.name
```

Comment:
This should also check if the selected dispvm is actually one of preloaded ones.
But also, what is the use case of pointing at specific DispVM, instead of its template?